docs/Authenticating-and-Authorizing-Users-via-keycloak.md

```diff
@@ -252,6 +253,85 @@ If all goes well, the following should happen:
 
 If this does not happen, see the Troubleshooting Tips  below.
 
+### Add client for OAuth2 token-based data access
+With cBioPortal instances that require user authentication the API can be queried when including a data access tokens in the request header (see [Authenticating Users via Tokens](Authenticating-Users-via-Tokens.md)). When using the `oauth2` implementation, KeyCloak can be configured as OAuth2 authentication provider that distributes data access tokens to users and validates these tokens when used while querying the API. This feature is enabled by creating a `cbioportal_api` OpenID Connect client that has access to the user roles defined in the `cbioportal` SAML client.
+
+1. Create a client with name `cbioportal_api`. Set _Client Protocol_ to `openid-connect`.
+
+![](images/previews/oauth2_client_1.png)
+
+2. On the configuration page of `cbioportal_api` client apply the following settings:
+
+#### Settings tab
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Access Type      | confidential |  |
+| Standard Flow Enabled      | ON      |   (default value) |
+| Enabled      | ON      |   (default value) |
+| Consent Required      | OFF        |   (default value) |
+| Client Protocol      | openid-connect   |   (default value) |
+| Access Type | confidential      |     (default value) |
+| Standard Flow Enabled | ON      |     (default value) |
+| Implicit Flow Enabled | OFF      |     (default value) |
+| Direct Access Grants Enabled | OFF      |    |
+| Service Accounts Enabled | OFF      |  (default value)   |
+| Authorization Enabled | OFF      |  (default value)  |
+| Valid Redirect URIs | _url_/api/data-access-token/oauth2  |  _url_ refers to base url of cBioPortal instance |
+
+![](images/previews/oauth2_client_3.png)
+
+#### Credentials tab
```

Comment:
This may be version specific - running keycloak 9, I don't see this tab

![image](https://user-images.githubusercontent.com/366003/76365140-efa3ee00-62fc-11ea-8f84-a15b19c3613f.png)

Reply:
I am running 8 and it is there...